design/gas/Design_gas.md

```diff
@@ -0,0 +1,72 @@
+# Obscuro Gas Design
+
+## Overview
+
+The revenue generation mechanism for Obscuro hinges on the collection of gas fees. Notably, as Obscuro operates as a layer 2 protocol, the cost of operations surpasses that of layer 1, since it has to cover layer 1 gas costs in addition to its operational expenses. Unlike layer 1 miners, who only shoulder operational costs, there's no expenditure prerequisite for block publication on Obscuro except for the static cost of stake.
+
+
+### Initial Bootstrapping of Network Gas
+
+Similar to other layer 2 protocols, Obscuro seeks to leverage Ethereum for gas usage. To facilitate this, Ethereum must first be bridged over to Obscuro. Although the bridge typically operates with a relayer, relaying a message incurs a gas cost. Hence, to bootstrap the entire layer 2 protocol, including relayers and bridges, cross-chain messages concerning Ethereum depositing into an account will be automatically executed on layer 2 at no cost and without the need for relaying. This ensures no additional transactions that necessitate publishing to layer 1.
+    
+
+### Types of Transactions
+
+Ethereum has developed to accommodate several [transaction types](https://docs.infura.io/networks/ethereum/concepts/transaction-types):
+1) Regular transactions - they work based on a gas auction.
+2) Access list transactions - same as previous ones, but they also announce what storage access they are supposed to perform when executed.
+3) EIP-1559 - the new transactions that burn the gas fee and work with tips in order to achieve a fair system.
+
+Usually layer 2's support EIP-1559 so there is a pending question of what we would need to support. 
+
+
+## Specification
+
+1. Ethereum must be used for gas operations and bridged from the layer 1 protocol.
+2. Bridged gas should automatically deposit into the respective addresses.
+3. Gas-consuming transactions should include a layer 1 cost component for calldata publishing and a layer 2 component that covers execution costs and prevents endless transaction execution.
+4. L2 gas price should be configurable. Furthermore it should support being free whilst still not allowing non halting execution.
+
```

Comment:
need more requirements here:

5.  The gas mechanics needs to prevent denial of service attacks.